libethcore/Precompiled.cpp

```diff
@@ -99,48 +99,77 @@ ETH_REGISTER_PRECOMPILED(identity)(bytesConstRef _in)
 
 // Parse _count bytes of _in starting with _begin offset as big endian int.
 // If there's not enough bytes in _in, consider it infinitely right-padded with zeroes.
-bigint parseBigEndianRightPadded(bytesConstRef _in, size_t _begin, size_t _count)
+bigint parseBigEndianRightPadded(bytesConstRef _in, bigint _begin, bigint _count)
 {
-	if (_begin > _in.count())
+	if (_begin > numeric_limits<size_t>::max())
+		return 0; // _begin overflows the range of size_t
+	if (_count > numeric_limits<size_t>::max())
+		return 0; // _begin overflows the range of size_t
+
+	size_t const begin{_begin};
+	size_t const count{_count};
+
+	if (begin > _in.count())
 		return 0;
 
 	// crop _in, not going beyond its size
-	bytesConstRef cropped = _in.cropped(_begin, min(_count, _in.count() - _begin));
+	bytesConstRef cropped = _in.cropped(begin, min(count, _in.count() - begin));
 
 	bigint ret = fromBigEndian<bigint>(cropped);
 	// shift as if we had right-padding zeroes
-	ret <<= 8 * (_count - cropped.count());
+	ret <<= 8 * (count - cropped.count());
```

Comment:
Right-hand operand to shift is not a bigint, it's probably `size_t`
This makes sense because shifting to more that `size_t` max would require extraordinary amount of memory